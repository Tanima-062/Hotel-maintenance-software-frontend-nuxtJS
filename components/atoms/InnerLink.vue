<template>
  <nuxt-link
    class="cursor-pointer focus:outline-none"
    :class="styles"
    :to="href"
  >
    <slot />
  </nuxt-link>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  href!: string

  @Prop({ default: 'default' })
  textColor!: string

  @Prop({ default: 'default' })
  hoverTextColor!: string

  @Prop({ default: true })
  showUnderline!: boolean

  @Prop({ default: true })
  focusOn!: boolean

  get styles(): string {
    const classList = []
    if (this.showUnderline) {
      classList.push('underline')
    }
    classList.push(`${this.textColor}`)
    classList.push(`hover:${this.hoverTextColor}`)
    classList.push(this.focusOn ? 'focus:shadow-outline' : 'focus:shadow-none')
    return classList.join(' ')
  }
}
</script>
