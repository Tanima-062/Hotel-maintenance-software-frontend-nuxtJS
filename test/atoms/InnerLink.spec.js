import { shallowMount } from '@vue/test-utils'
import InnerLink from '@/components/atoms/InnerLink.vue'

describe('Atoms InnerLink Check', () => {
  test('InnerLink Test', () => {
    const wrapper = shallowMount(InnerLink, {
      stubs: {
        NuxtLink: true,
      },
    })

    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
