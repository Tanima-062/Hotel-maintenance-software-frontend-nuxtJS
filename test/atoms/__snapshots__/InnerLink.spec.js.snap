// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms InnerLink Check InnerLink Test 1`] = `
<nuxtlink-stub
  class="cursor-pointer focus:outline-none underline default hover:default focus:shadow-outline"
  to=""
/>
`;
